adam-core/src/main/scala/org/bdgenomics/adam/converters/FastqRecordConverter.scala

```diff
@@ -143,10 +145,25 @@ class FastqRecordConverter extends Serializable with Logging {
     // get fields for first read in pair
     val readName = trimTrailingReadNumber(lines(0).drop(1))
     val readSequence = lines(1)
-    val readQualities = lines(3)
+
+    if (stringency == ValidationStringency.STRICT && lines(3) == "*" && readSequence.length > 1)
+      throw new Exception(s"Fastq quality must be defined")
+    else if (stringency == ValidationStringency.STRICT && lines(3).length != readSequence.length)
+      throw new Exception(s"Fastq sequence and quality strings must have the same length")
+
+    val readQualities =
```

Comment:
Can you be more specific with your proposal?  Are you suggesting adding
another parameter globally, or just for FQ?



Sent with MailTrack
<https://mailtrack.io/install?source=signature&lang=en&referral=allenday@allenday.com&idSignature=22>

On Tue, Nov 3, 2015 at 4:12 AM, Michael L Heuer <notifications@github.com>
wrote:

> In
> adam-core/src/main/scala/org/bdgenomics/adam/converters/FastqRecordConverter.scala
> <https://github.com/bigdatagenomics/adam/pull/856#discussion_r43675511>:
>
> > @@ -143,10 +145,25 @@ class FastqRecordConverter extends Serializable with Logging {
> >      // get fields for first read in pair
> >      val readName = trimTrailingReadNumber(lines(0).drop(1))
> >      val readSequence = lines(1)
> > -    val readQualities = lines(3)
> > +
> > +    if (stringency == ValidationStringency.STRICT && lines(3) == "*" && readSequence.length > 1)
> > +      throw new Exception(s"Fastq quality must be defined")
> > +    else if (stringency == ValidationStringency.STRICT && lines(3).length != readSequence.length)
> > +      throw new Exception(s"Fastq sequence and quality strings must have the same length")
> > +
> > +    val readQualities =
>
> Emit warnings but keep going if possible and fix invalid input data are
> slightly different things. I think with a parameter or parameters presented
> to the user I'd support this change.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bigdatagenomics/adam/pull/856/files#r43675511>.
>